### PR TITLE
Feature/61 button pub

### PR DESCRIPTION
**Pull request by jamesmelzer**
* created the buttons doc page
* opened up the spacing around label on horizontal component displays 
* added our first doc illustrations (confirm folder structure: doc-site/images/[featurename]/)
* moved typography and elevation sink pages from /visual-styles/ to /visual-style/

@joesnellpdx Is there much to document behind the "loading" state on some buttons? I'm not sure what it does, so I didn't document it...

Big kudos to @kevinmpowell - This doc thoroughly worked out the code example pairs and the code reference table. Both worked well and were sufficiently documented. Thanks!

### CODE REVIEW

File: doc-site/docs/components/buttons.njk [hunk 1]
```diff
@@ -1,204 +1,601 @@
...
+    code_snippet_source=(
+      library.button(loader="true", text="Submit Donation") + 
+      library.button(class="spirit-button--loading", loader="true", text="Submit Donation") + 
+      library.button(loader="true", feedbackType="true", text="Submit Donation")
+    )
+  )
+  }}
+  {% filter markdown %}
+
+      <div class="spirit-alert spirit-alert--info">Note: These button states are presented without Javascript. Product teams should wire up these visual states using their own application logic. </a>
```
Comment (joesnellpdx) [inline doc-site/docs/components/buttons.njk]: @jamesmelzer 

_Suggestion:_
`Note: These button states are presented without Javascript. Product teams should wire up these visual states using their own application logic by altering the aria-hidden="" values within the button component.`

**Comment by patrickmckowen** [comment]
@jamesmelzer overall looks good!  A few thoughts:

1. the Page Title feels small.  I tried moving it inside the Long Form Text container and that changed the size appropriately.
2. Sections of content are running together e.g. [this](https://gmkr.io/s/5b6b206a3c82662695192e54/0) and [this](https://gmkr.io/s/5b6b20df3c82662695192e55/0).  Have you tried using 12x and 24px bottom margin to air out the different sections? I wonder if this would support scanning and stay true to Spirit's generous use of negative space to create structure.

I also noticed that the long form text paragraph uses black instead of dark grey.  When we update that it should also help with hierarchy.

**Comment by jamesmelzer** [comment]
@patrickmckowen 
There is some inconsistency with the way long form text is bumping up against other components versus spacing itself internally. We'll want to normalize that. Hopefully it will fix what you're seeing. 

h2 with generous 64px after component
h2 follows component

h2 with tight 32px after paragraph (due to collapsed margins within LFT)
h2 follows paragraph

Also, which grey are you thinking for body text? It's currently `$spirit-color-black` #00081A

File: doc-site/docs/components/buttons.njk [hunk 2]
```diff
@@ -1,204 +1,599 @@
...
-      {
-        example: library.button(class="spirit-button--icon-only spirit-button--hover", icon="spirit-chevron-right"),
-        label: "Hover"
-      },
-      {
-        example: library.button(class="spirit-button--icon-only spirit-button--active", icon="spirit-chevron-right"),
-        label: "Active"
-      },
+
+  {% set button_state_example_icon_only %}
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: screen shot 2018-08-09 at 11 03 46 am

* [ ] Center icon only buttons and labels

File: doc-site/docs/components/buttons.njk [hunk 3]
```diff
@@ -1,204 +1,599 @@
...
-      ### Small
+      ### Toggle Button
+      Use in place of a radio group or checkbox group to capture a key user selection. 
+
+      #### Using Checkboxes to Show Toggle States
   {% endfilter %}
 
-  {% set button_small %}
-    {{ library.button(class="spirit-button--small", text="Learn More") }}
+  {% set button_state_example_toggle %}
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: screen shot 2018-08-09 at 11 05 41 am

* [ ] Center "Active/Selected" label

File: doc-site/docs/components/buttons.njk (hunk 1 above)
Comment (kevinmpowell) [reply doc-site/docs/components/buttons.njk]: @joesnellpdx I don't think the `aria-hidden` attribute is what controls these visual states. It's modifier classes correct?

File: doc-site/docs/components/buttons.njk [hunk 4]
```diff
@@ -1,204 +1,599 @@
...
 
+  {{ esds_doc.code_example_pair(
+    examples=[ { example: button_state_example_toggle_radio } ],
+    code_snippet_source=(button_state_example_toggle_radio)
+    )
+  }}
+
+  {% filter markdown %}
+
+      * Apply toggle button to a radio group to capture a selection among mutually-exclusive choices. This is the default control for capturing a donation amount, as depicted above.
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: "Apply toggle button" is unclear. What is being applied?
I don't think this is a simple re-skinning of the radio group and checkbox group components, I believe these are bespoke button components. I think your statement earlier where you say "Use in place of a standard radio/checkbox group" is clearer.

File: doc-site/docs/components/buttons.njk [hunk 5]
```diff
@@ -1,204 +1,599 @@
...
+      library.button(class="spirit-button--success", text="Success") + 
+      library.button(class="spirit-button--danger", text="Delete") + library.button(class="spirit-button--success", iconName="check", iconLabel="Success")
+    )
+  )
   }}
-
   {% filter markdown %}
-      ### Small
+      ### Toggle Button
+      Use in place of a radio group or checkbox group to capture a key user selection.
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: "Checkbox Group" is too specific. A checkbox toggle can be used standalone. 
"Use in place of radio buttons or checkboxes..." is broader and still communicates "use when."

File: doc-site/docs/components/buttons.njk [hunk 6]
```diff
@@ -1,204 +1,599 @@
...
+  {{ esds_doc.html_example_list( example=button_size_example_extrasmall ) }}
 
   {% filter markdown %}
-      ### Large
+      #### Small
   {% endfilter %}
 
-  {% set button_large %}
-    {{ library.button(class="spirit-button--large", text="Learn More") }}
+  {% set button_size_example_small %}
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: screen shot 2018-08-09 at 11 12 04 am

* [ ] Center and bottom align "Icon Only" labels for small and default examples

File: doc-site/docs/components/buttons.njk [hunk 7]
```diff
@@ -1,204 +1,599 @@
...
+      
+      Where feasible, use `<button>`. For applications that generate markup beyond your control, using `<input type=”submit”>` is also acceptable. 
+      
+      ### Class Reference
+  {% endfilter %}
+
+    {{ esds_doc.reference_table(column_widths=[35, 30, 35],
+    data=[
+      {
+        class: '.spirit-button',
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: * [ ] Remove `.spirit-button` from class reference. Show elsewhere, via code examples, how a button can be one of `<a>`, `<button>`, or `<input>`.
Simply applying the class isn't sufficient, the markup changes as well depending on the underlying element. Having this in the table is misleading.

In general, only put modifier (`--something`) classes in the reference tables. Adding block and element classes to the tables isn't something that scales well across components.

File: doc-site/docs/components/buttons.njk [hunk 8]
```diff
@@ -1,204 +1,599 @@
...
+        applies_to: '.spirit-button',
+        outcome: 'Applies icon only button style, which includes conversion to a round shape.'
+      },
+      {
+        class: '.spirit-button--toggle',
+        applies_to: '.spirit-button',
+        outcome: 'Applies toggle button style to `<input type=”radio”>` or `<input type=”checkbox”>`'
+      },
+      {
+        class: '.spirit-button--small',
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: screen shot 2018-08-09 at 11 17 03 am


* [ ] Remove these last few rows from the table, they're covered by the Sizes and Responsive sizing sections above and will be accessible via the anchor nav. If desired, add a note at the bottom of the table with an anchor link back up to those sections.

File: doc-site/docs/components/buttons.njk [hunk 9]
```diff
@@ -1,204 +1,599 @@
...
+
   {% endset %}
-  {{ esds_doc.code_example_pair( example=button_large ) }}
+  {{ esds_doc.html_example_list( example=button_size_example_large ) }}
+
+  {% filter markdown %}
+  
+      ### Responsive Sizing
+      Buttons can be sized responsively. By default, buttons stay the same size on all viewports. By adding breakpoint-specific classes, a button can have different sizes for small, medium, and large viewports. 
+
```
Comment (kevinmpowell) [inline doc-site/docs/components/buttons.njk]: * [ ] Add the modifier class pattern: `.spirit-button--[size]-[breakpoint]` and then provide code snippets with a paragraph summarizing the outcome.

A class of:
```
.spirit-button--extra-small-lg
```
Would result in an extra small button at the largest viewport breakpoint.

File: doc-site/components/code_example_pair/code_example_pair.scss [hunk 10]
```diff
@@ -61,6 +61,10 @@ $esds-doc-code-snippet-space-stack: $spirit-space-generic-4-x;
   }
 }
 
+.esds-doc-code-example-pair__example-list {
```
Comment (kevinmpowell) [inline doc-site/components/code_example_pair/code_example_pair.scss]: * [ ] remove `overflow: hidden;` from code_example_pair. 

We're going to want examples like tooltips and popovers to potentially overflow the example list. 

I'm curious, why did you need this?

Comment (jamesmelzer) [reply doc-site/components/code_example_pair/code_example_pair.scss]: Background color doesn't respect the rounded corners without it. I'll do a first-child approach instead.

Comment (jamesmelzer) [reply doc-site/components/code_example_pair/code_example_pair.scss]: Worth noting, I ended up using default background in my first example in every case, but that's a coincidence...

File: doc-site/docs/components/buttons.njk (hunk 5 above)
Comment (jamesmelzer) [reply doc-site/docs/components/buttons.njk]: Fixed.

File: doc-site/docs/components/buttons.njk (hunk 4 above)
Comment (jamesmelzer) [reply doc-site/docs/components/buttons.njk]: Fixed.

File: doc-site/docs/components/buttons.njk (hunk 3 above)
Comment (jamesmelzer) [reply doc-site/docs/components/buttons.njk]: This felt a little brute force, but it's only buttons inside...
  & > *:first-child {
    display: block;
    margin: 0 auto;
  }

**Comment by jamesmelzer** [comment]
@kevinmpowell Thanks for feedback! Ready for another look.

**Comment by joesnellpdx** [comment]
@kevinmpowell @jamesmelzer 

Current setup for loading button:

1) Has class `.spirit-button--loading`: `.spirit-button__icon-contain` has aria-hidden="false", `.spirit-button__default-text` has aria-hidden="true", `.spirit-button__success-text` or `spirit-button__icon-contain` have aria-hidden="true"
2) No class `.spirit-button--loading` and is "success": `.spirit-button__icon-contain` has aria-hidden="true", `.spirit-button__default-text` has aria-hidden="true", `.spirit-button__success-text` or `spirit-button__icon-contain` have aria-hidden="false"
3) No class `.spirit-button--loading`: `.spirit-button__icon-contain` has aria-hidden="false", `.spirit-button__default-text` has aria-hidden="true", `.spirit-button__success-text` or `spirit-button__icon-contain` have aria-hidden="true"`

All above need attribute: `data-loader=["true"]`
Can not be `<input>`

**Comment by kevinmpowell** [comment]
Fix sass-lint warnings:
screen shot 2018-08-09 at 2 35 26 pm

**Comment by kevinmpowell** [comment]
screen shot 2018-08-09 at 2 37 09 pm
* [ ] Normalize spacing between these two examples

**Comment by kevinmpowell** [comment]
screen shot 2018-08-09 at 2 39 44 pm

The additional `:first-child` and `:last-child` padding on the html example list items is making these button examples look vertically misaligned. @jamesmelzer what do you think about removing the additional padding from `:first-child` and `:last-child` in the code_example_pair?

Obviously this can be done as part of a separate issue.

**Comment by jamesmelzer** [comment]
@kevinmpowell Let's leave the first-child/last-child padding in place until we have a few more of these pages published. It can look really tight without it, and I still think the mild off-centered alignment is preferable to the overall tightness. Example... 
screen shot 2018-08-09 at 3 48 32 pm
